java/src/humbug/HumbugConsent.java

```diff
@@ -0,0 +1,24 @@
+package humbug;
+
+
+public class HumbugConsent {
+    private final boolean mechanism;
+    private String envVar;
+    private String[] optOutValues;
+
+    public HumbugConsent() {
+        this(false);
+    }
+
+    public HumbugConsent(boolean mechanism) {
+        this.mechanism = mechanism;
+
+        this.envVar = "BUGGER_OFF";
+        this.optOutValues = new String[]{"yes"};
+    }
+    public boolean check() {
```

Comment:
Behavior `HumbugConsent` is as follows:

HumbugConsent can be instantiated with a *list* of consent mechanisms. Each consent mechanism is either a boolean or a function with no arguments which returns a boolean value.

To resolve the consent, on `check`, the `HumbugConsent` object:
1. Checks if the `BUGGER_OFF` environment variable has been set to `true` or `1` or `y` or any other value signifying "yes" (check Python version of library for implementation). If so, it returns false.
2. Goes through the list item by item. If the item is boolean, it checks if it is false. If the item is a callable, it calls it and checks if the return value is false. If it encounters a false value at any step, it returns false. Else, returns true.